Add desktop samples to regression samples and fix bugs

inforithmics opened the pull request.

pauldendulk commented on Mapsui/Cache/IUrlPersistentCache.cs:
```diff
@@ -0,0 +1,8 @@
+﻿namespace Mapsui.Cache;
+
+public interface IUrlPersistentCache
```
Nice generic solution. Perhaps we could even move towards a fully generic solution. Cache everything (tiles and json) in under a url key. Like how it works in browser caching. Caching could be done in a wrapper around the HttpClient. Then again you may want to use a specific strategies for tiles.

inforithmics replied:
I like to have the tiles cached differently because there is an index over the tiles and that uses only ints which is much faster than using string index and concatenation. But maybe this can be tested with a benchmark if the tile index is really faster than a string.

pauldendulk replied:
It is possible to have different instances of HttpClient with different strategies ([named clients](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/http-requests?view=aspnetcore-6.0#named-clients)). But then, that is far removed from where we are now. We are not even using the HttpClientFactory.

inforithmics commented on Mapsui/Extensions/StreamExtensions.cs:
```diff
@@ -8,10 +10,25 @@ public static class StreamExtensions
+                        break;
+                }
+            }
+            catch (NotSupportedException)
```
removed it is not necessary was  a workaround because of ContentLengthReadStream. I handle this now.

inforithmics commented on Mapsui/Extensions/StreamExtensions.cs:
```diff
@@ -8,10 +10,25 @@ public static class StreamExtensions
-            if (input.Position != 0)
+            try
+            {
+                switch (input.GetType().Name)
```
Is a private class so I can use only the name for detecting this. Causes an exption when I ask the position

inforithmics commented on Mapsui.UI.Eto/Mapsui.UI.Eto.csproj:
```diff
@@ -9,7 +9,7 @@
   <ItemGroup>
     <PackageReference Include="Eto.SkiaDraw" Version="0.2.0" />
-    <PackageReference Include="System.Text.Encoding.CodePages" Version="5.0.0" />
+    <PackageReference Include="System.Text.Encoding.CodePages" Version="6.0.0" />
```
removed it

inforithmics commented:
1. I made Bugfixes in the Tests should be more reliable (Retry 5 Times). 
2. Fixed an Bug in the Animation (It said it has Animation running but was stopped)
3. Fixed CompareImage if the Images have different Sizes (Caused Memory Corruption issue).

inforithmics commented on Mapsui.Nts/Providers/Shapefile/ShapeFile.cs:
```diff
@@ -129,6 +130,20 @@ public enum ShapeType
+        {
+            try
+            {
+                // Workaround for Bug in Shapefile
```
// Without this Fix this method throws an exception:
                // Encoding.GetEncoding(...)
                // System.NotSupportedException: 'No data is available for encoding 1252. For information on defining a custom encoding, see the documentation for the Encoding.RegisterProvider method.'
                // StackOverflow
                // https://stackoverflow.com/questions/50858209/system-notsupportedexception-no-data-is-available-for-encoding-1252

This happens in Mobile Clients and NET6

pauldendulk replied:
After I posted that question I ran into this bug myself for some project. Could be solved by locally setting 'Encoding.RegisterProvider(CodePagesEncodingProvider.Instance);', but yes we should prevent this.

pauldendulk reviewed (approved):
Great! this is something that will benefit us in the long run.
